ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -82,71 +82,126 @@ public void copyAndVerify(FileSystem destinationFs, Path destRoot,
           LOG.error("Failed to create destination directory: " + destination);
           throw new IOException("Destination directory creation failed");
         }
-        doCopyRetry(sourceFs, fileInfoList, destinationFs, destination, useRegularCopy);
 
-        // Verify checksum, retry if checksum changed
-        List<ReplChangeManager.FileInfo> retryFileInfoList = new ArrayList<>();
-        for (ReplChangeManager.FileInfo srcFile : srcFiles) {
-          if (!srcFile.isUseSourcePath()) {
-            // If already use cmpath, nothing we can do here, skip this file
-            continue;
-          }
-          String sourceChecksumString = srcFile.getCheckSum();
-          if (sourceChecksumString != null) {
-            String verifySourceChecksumString;
-            try {
-              verifySourceChecksumString
-                      = ReplChangeManager.checksumFor(srcFile.getSourcePath(), sourceFs);
-            } catch (IOException e) {
-              // Retry with CM path
-              verifySourceChecksumString = null;
-            }
-            if ((verifySourceChecksumString == null)
-                    || !sourceChecksumString.equals(verifySourceChecksumString)) {
-              // If checksum does not match, likely the file is changed/removed, copy again from cm
-              srcFile.setIsUseSourcePath(false);
-              retryFileInfoList.add(srcFile);
-            }
-          }
-        }
-        if (!retryFileInfoList.isEmpty()) {
-          doCopyRetry(sourceFs, retryFileInfoList, destinationFs, destination, useRegularCopy);
-        }
+		    // Copy files with retry logic on failure
```

Comment:
retry logic on failure and source file modification

Reply:
Will fix this if any code change needed for this patch. :)

Reply:
done